frontend/src/app/modules/leader/leader-board/leader-board.component.ts

```diff
@@ -56,19 +90,131 @@ export class LeaderBoardComponent extends BaseComponent implements OnInit {
             .pipe(takeUntil(this.unsubscribe$))
             .subscribe({
                 next: (users) => {
+                    this.handleUsersUpdate(users);
+                },
+                error: () => {
                     this.loading = false;
-                    if (!users.length) {
-                        this.isLastPage = true;
+                    this.toastrNotification.showError('Server connection error');
+                },
+            });
+    }
 
-                        return;
-                    }
-                    this.users = [...this.users, ...users];
-                    this.usersToShow = this.users;
+    private getFriends() {
+        if (this.isLastFriendsPage) {
+            return;
+        }
+
+        this.friendsPage.pageNumber++;
+        this.loading = true;
+
+        this.userService
+            .getFriendsLeaderBoard(this.friendsPage)
+            .pipe(takeUntil(this.unsubscribe$))
+            .subscribe({
+                next: (users) => {
+                    this.handleFriendsUpdate(users);
                 },
                 error: () => {
                     this.loading = false;
                     this.toastrNotification.showError('Server connection error');
                 },
             });
     }
+
+    private getCurrentUser() {
```

Comment:
see the previous comments. almost the same as getFriends

Reply:
https://github.com/BinaryStudioAcademy/bsa-2023-leetwars/pull/187#discussion_r1335256788